nixops/backends/__init__.py

```diff
@@ -242,9 +242,9 @@ def public_ipv4(self):
     def private_ipv4(self):
         return None
 
-    def address_to(self, m):
-        """Return the IP address to be used to access machone "m" from this machine."""
-        ip = m.public_ipv4
+    def address_to(self, r):
+        """Return the IP address to be used to access resource "r" from this machine."""
+        ip = getattr(r, 'public_ipv4', None)
```

Comment:
Why not move the public_ipv4 property to the resource class? Calling `getattr` from the outside seems ugly.